api/controllers/projectGroupV2.js

```diff
@@ -0,0 +1,303 @@
+{
+    defaultLog.debug('>>> {GET} /Projects/{projId}/Groups/{groupId}/members');
+
+    try
+    {
+        if (args.swagger.params.hasOwnProperty('projId') && args.swagger.params.hasOwnProperty('groupId'))
+        {
+            let projectId = args.swagger.params.projId.value;
+            let groupId = args.swagger.params.groupId.value;
+
+            defaultLog.debug(' Fetching group ' + groupId + ' members from project ' + projectId);
+
+            let project = await projectDAO.getProject(SECURE_ROLES, projectId);
+            let group = await projectGroupDAO.getGroup(groupId);
+
+            if(project && group)
+            {
+                let pageNumber = args.swagger.params.hasOwnProperty('pageNumber') && args.swagger.params.pageNumber.value ? args.swagger.params.pageNumber.value : 1;
+                let pageSize   = args.swagger.params.hasOwnProperty('pageSize')   && args.swagger.params.pageSize.value   ? args.swagger.params.pageSize.value   : 10;
+                let sortBy     = args.swagger.params.hasOwnProperty('sortBy')     && args.swagger.params.sortBy.value     ? args.swagger.params.sortBy.value     : '';
+
+                let members = await projectGroupDAO.getGroupMembers(SECURE_ROLES, args.swagger.params.auth_payload.preferred_username, group, sortBy, pageSize, pageNumber);
+                members = projectGroupDAO.groupHateoas(members, SECURE_ROLES);
+                return Actions.sendResponseV2(res, 200, members);
+            }
+            else
+            {
+                return Actions.sendResponseV2(res, 404, { status: 404, message: 'Project ' + projectId + ' or Group ' + groupId + ' not found.'});
+            }
+        }
+        else
+        {
+            throw Error('Invalid request');
+        }
+    }
+    catch (e)
+    {
+        defaultLog.error('### Error in {GET} /Projects/{projId}/Groups/{groupId}/members :', e);
+        return Actions.sendResponseV2(res, 500, { code: '500', message: 'Internal Server Error', self: 'Api/Projects{id}/Groups/{id}/members' });        
+    }
+    finally
+    {
+        defaultLog.debug('<<< {GET} /Projects/{projId}/Groups/{groupId}/members');
+    }
+};
+
+// DELETE (Protected only deleteGroupMember)
+exports.deleteGroupMember = async function (args, res, rest) 
+{
+    defaultLog.debug('>>> {DELETE} /Projects/{projId}/Groups/{groupId}/members/{memberId}');
+
+    try
+    {
+        if (args.swagger.params.hasOwnProperty('projId') && args.swagger.params.hasOwnProperty('groupId') && args.swagger.params.hasOwnProperty('memberId'))
+        {
+            let projectId = args.swagger.params.projId.value;
+            let groupId = args.swagger.params.groupId.value;
+            let memberId = args.swagger.params.memberId.value;
+
+            defaultLog.debug(' Deleting member ' + memberId + ' from group ' + groupId + ' from project ' + projectId);
+
+            let project = await projectDAO.getProject(SECURE_ROLES, projectId);
```

Comment:
Can we initiate these at the same time and use Promise.all()?

Reply:
Yep, that shouldn't be a problem. I'll review through some other spots and update where we're waiting for synchronous awaits as well.